runelite-api/src/main/java/net/runelite/api/events/EventBus.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2018, Tomas Slusny <slusnucky@gmail.com>
+ * Copyright (c) 2018, Abex
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.api.events;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.eventbus.Subscribe;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class EventBus
+{
+	@Value
+	private static class Subscriber
+	{
+		private final Object object;
+		private final Method method;
+
+		void invoke(final Object arg) throws Exception
+		{
+			method.invoke(object, arg);
+		}
+	}
+
+	private ImmutableMultimap<Class, Subscriber> subscribers;
+
+	public synchronized void register(final Object object)
+	{
+		final ImmutableMultimap.Builder<Class, Subscriber> builder = ImmutableMultimap.builder();
+
+		if (subscribers != null)
+		{
+			builder.putAll(subscribers);
+		}
+
+		for (Class<?> clazz = object.getClass(); clazz != null; clazz = clazz.getSuperclass())
+		{
+			for (final Method method : clazz.getDeclaredMethods())
+			{
+				final Subscribe sub = method.getAnnotation(Subscribe.class);
+
+				if (sub == null)
+				{
+					continue;
+				}
+
+				assert method.getReturnType() == Void.TYPE : "@Subscribed methods cannot return a value";
```

Comment:
It might be better to use `IllegalArgumentException`

Reply:
Replaced with Preconditions.checkArgument